engine/server/buildkitcontroller.go

```diff
@@ -37,6 +37,7 @@ import (
 	"github.com/moby/buildkit/util/throttle"
 	"github.com/moby/buildkit/util/tracing/transform"
 	bkworker "github.com/moby/buildkit/worker"
+	"github.com/moby/locker"
```

Comment:
This is https://github.com/moby/locker, which is used in https://github.com/moby/moby, as well as in upstream https://github.com/moby/buildkit in various places.